payment/src/test/java/org/killbill/billing/payment/api/TestPaymentApi.java

```diff
@@ -320,6 +320,23 @@ public void testCreatePurchaseWithControlPaymentPluginException() throws Excepti
 
     @Test(groups = "slow")
     public void testCreatePurchaseWithControlPluginException() throws Exception {
+        mockPaymentControlProviderPlugin.throwsException(new PaymentControlApiException());
+
+        final BigDecimal requestedAmount = BigDecimal.TEN;
+        final String paymentExternalKey = "pay controle external key";;
+        final String transactionExternalKey = "txn control external key";
+        try {
+            paymentApi.createPurchaseWithPaymentControl(
+                    account, account.getPaymentMethodId(), null, requestedAmount, Currency.AED,
+                    paymentExternalKey, transactionExternalKey, ImmutableList.<PluginProperty>of(), CONTROL_PLUGIN_OPTIONS, callContext);
+            fail();
+        } catch (PaymentApiException e) {
+            assertTrue(e.getCause() instanceof PaymentControlApiException);
+        }
+    }
+
```

Comment:
Good test. I am assuming the goal was *not* to test the aborted state here but the case where plugin throws.